apps/reaper/lib/reaper/data_extract/extract_step.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Reaper.DataExtract.ExtractStep do
+  @moduledoc """
+  This module processes extract steps as defined in a dataset definition.  After
+  iterating through the steps, accumilating any destination values in the assigns block
+  it is assumed the final step will be http (at this time) which returns a data stream
+  """
+  require Logger
+  alias Reaper.DataSlurper
+  alias Reaper.UrlBuilder
+  alias Reaper.Decoder
+
+  def execute_extract_steps(dataset, steps) do
+    Enum.reduce(steps, %{}, fn step, acc ->
+      execute_extract_step(dataset, step, acc)
+    end)
+  end
+
+  defp execute_extract_step(dataset, step, assigns_accumulator) do
+    step = Map.put(step, :assigns, Map.merge(step.assigns, assigns_accumulator))
```

Comment:
wait, is this not the equivalent of
```elixir
Map.merge(step, %{assigns: assigns_accumulator})
```
?

Reply:
Wrong!